### DISCUSSION TITLE
CC: Tweaked 1.97.0

### DISCUSSION THREAD

**Discussion by SquidDev** [cc-tweaked/CC-Tweaked#843] [Announcements]
I'm currently running my release script, which must mean that another release is immanent! [Unlike 1.96.0][previous], CC: Tweaked 1.97.0 has a couple of exciting features, as well as a whole slew of bug fixes. Anyway, let's get into the highlights:

### Computer GUI buttons
Computer GUIs (normal computers, but also turtles and pocket computers) now have a bit more UI! They now container a power button (which also doubles as a power indicator) and a button to terminate the current program.

![The computer GUI, with buttons on the left hand side](https://user-images.githubusercontent.com/4346137/123710138-1e353780-d866-11eb-98ae-05c64a8e55f2.png)

I'm hoping this'll help make the UI a little more discoverable - it's definitely not obvious how to shutdown a computer right now.

### File uploads (1.16.5 only)
It's now possible to upload a file by dragging and dropping it onto an open computer.

![A confirmation dialogue for file uploads](https://user-images.githubusercontent.com/4346137/123710845-54bf8200-d867-11eb-97b6-6fe5c09bdae1.png)

There's currently some rather strict upload limits (max 30KiB), but am hoping to loosen these in the future. It's hopefully enough for most small scripts!

I'd definitely like some feedback about this feature, especially on how to make it more discoverable - please leave a comment below (or start a new discussion) if you've any thoughts!

### Some other highlights
This release also:

 - Fixes one of the longest remaining known ComputerCraft bugs (https://github.com/dan200/ComputerCraft/issues/301)
 - Adds supports for MoreRed bundled cables (1.16.5 only)
 - Removes the config option for the debug API, meaning it is now guaranteed to be present.

As well as a whole bunch of other things. As always, please check the changelog.

### In closing

Many thanks to everyone who contributed to this release, especially Anavrins, Jummit, Naheulf, MCJack123, magiczocker10, fatboychummy, Wojbie and Ronan-H for their PRs!

As many of you will be aware Minecraft 1.17 is out, and Forge will be soon to follow. This means I'll be dropping support for 1.15.x relatively shortly as we update to the latest Minecraft version. 1.97.0 is probably the last feature release for 1.15.x (though there may still be some bugfixes to come) - judging by the download numbers, this shouldn't affect too many people though :).

[previous]: https://github.com/SquidDev-CC/CC-Tweaked/discussions/795

**Comment by Kan18** [comment]
You could make the drag-and-drop feature more discoverable by adding an entry to /rom/motd.txt

**Comment by SirEndii** [comment]
Wonderful ^^

**Comment by LogicEngineer** [comment]
How about some extra buttons labelled 1-9 or something that we could assign scripts to?

**Reply by SkyTheCodeMaster** [reply]
Something like this could possibly queue an event?
`"terminal_button", <button number>`

**Reply by fatboychummy** [reply]
Oh that'd be neat! Multishell could hook into it to swap tabs on basic computers.

**Reply by Lupus590** [reply]
Hmm, it would be cool if we could set a display character for these buttons. Multishell could set it as a next and previous arrow character then.
